DirectEmployers/MyJobs
# [PD-2959] add search functionality

## What's this PR do?
Allows for searching the current analytics tab.

## Where should the reviewer start?
This should work on any of the analytics charts.

## How should this be manually tested?
Go to any (or all!) of the analytics charts. Search for something that's on the current page. Search for something that's on another page. Search for something that isn't present at all.

All of them should provide exactly what you would expect from search results.

## Any background context you want to provide?
Search terms should only be applied to the current column displayed to the user (so if the user is looking at a list of job titles, search should only be done on the job titles).

Search results should update the history.

After reviewing how search was implemented, I decided to hold off on implementing search on the other datasources for now. I don't think the way search is implemented is necessarily applicable to other datasources, so implementing it now will just introduce a useless maintenance nightmare.
